packages/opentelemetry-sdk-trace-base/src/export/SimpleSpanProcessor.ts

```diff
@@ -57,21 +57,31 @@ export class SimpleSpanProcessor implements SpanProcessor {
       return;
     }
 
-    internal
-      ._export(this._exporter, [span])
-      .then((result: ExportResult) => {
-        if (result.code !== ExportResultCode.SUCCESS) {
-          globalErrorHandler(
-            result.error ??
-              new Error(
-                `SimpleSpanProcessor: span export failed (status ${result})`
-              )
-          );
-        }
-      })
-      .catch(error => {
-        globalErrorHandler(error);
-      });
+    const doExport = () =>
+      internal
+        ._export(this._exporter, [span])
+        .then((result: ExportResult) => {
+          if (result.code !== ExportResultCode.SUCCESS) {
+            globalErrorHandler(
+              result.error ??
+                new Error(
+                  `SimpleSpanProcessor: span export failed (status ${result})`
+                )
+            );
+          }
+        })
+        .catch(error => {
+          globalErrorHandler(error);
+        });
+
+    // Avoid scheduling a promise to make the behavior more predictable and easier to test
+    if (span.resource.asyncAttributesHaveResolved()) {
+      void doExport();
+    } else {
+      span.resource
+        .waitForAsyncAttributes()
+        .then(doExport, err => globalErrorHandler(err));
```

Comment:
for `BatchSpanProcessor`, when the promise fail we
```js
diag.debug('Error while resolving async portion of resource: ', err)
```

and here its `globalErrorHandler`

I suggest choosing one and making it consistent